paddle/fluid/operators/math/pooling.cc

```diff
@@ -712,10 +720,9 @@ class Pool3dFunctor<platform::CPUDeviceContext, PoolProcess, T> {
                   }
                 }
               }
-              int pool_size =
-                  (exclusive || adaptive)
-                      ? (dend - dstart) * (hend - hstart) * (wend - wstart)
-                      : ksize_depth * ksize_height * ksize_width;
+              if (exclusive || adaptive) {
```

Comment:
this check seems redundant now, isn't `pool_size` always the same value?

Reply:
It is not redundant to update`pool_size`  when `padding ` is not zero.
The variable `pool_size` will be updated after running the following code.
```
pool_size = (dend - dstart) * (hend - hstart) * (wend - wstart);
dstart = std::max(dstart, 0);
hstart = std::max(hstart, 0);
wstart = std::max(wstart, 0);
dend = std::min(dend, input_depth);
hend = std::min(hend, input_height);
wend = std::min(wend, input_width);
```